testing/idris/APKBUILD

```diff
@@ -51,30 +49,34 @@ prepare() {
 	cd "$builddir"
 
 	# Create cabal config and fetch index.
-	HOME="${CABAL_CONFIG%/*}" cabal update
+	HOME="$CABAL_HOME" cabal update
 
 	# Cabal implicitly loads cabal.config from the project's directory.
 	ln -sf "$srcdir"/cabal.config .
 
 	# Create Cabal sandbox that will be used for all subsequent cabal
 	# invocations. This is something like Python virtual env.
-	cabal sandbox init
+	HOME="$CABAL_HOME" cabal sandbox init
 
 	# Fetch all dependencies.
-	# XXX: This should always fetch the same versions as defined in
-	# our cabal.config, but Cabal does not verify packages integrity!
-	cabal fetch libffi .
+	for pkg in $(cat cabal.config | sed -e 's/constraints[:]//g' -e 's/--.*//g' | tr -d '[:space:]' | sed -e 's/[,]/ /g' -e 's/==/-/g'); do
```

Comment:
Ok split it apart into 3 functions as I noticed my second copy/paste was wrong, now it actually passes --constraint=package==version properly. It worked more due to dumb luck before but this is what I was trying to accomplish to tell the cabal install constraint solver to use the freeze packages and not what the newest that might be in the index.